packages/core/src/node/messaging/ipc-protocol.ts

```diff
@@ -8,7 +8,41 @@
 
 import { MessageConnection } from "vscode-jsonrpc";
 
-export const THEIA_PARENT_PID = 'THEIA_PARENT_PID';
-export const THEIA_ENTRY_POINT = 'THEIA_ENTRY_POINT';
+const THEIA_PARENT_PID = 'THEIA_PARENT_PID';
+const THEIA_ENTRY_POINT = 'THEIA_ENTRY_POINT';
 
 export type IPCEntryPoint = (connection: MessageConnection) => void;
+
+/**
+ * Exit the current process if the parent process is not alive.
+ * Relevant only for some OS, like Windows
+ */
+export function checkParentAlive(): void {
+    if (process.env[THEIA_PARENT_PID]) {
+        const parentPid = Number(process.env[THEIA_PARENT_PID]);
+
+        if (typeof parentPid === 'number' && !isNaN(parentPid)) {
+            setInterval(function () {
+                try {
+                    // throws an exception if the main process doesn't exist anymore.
+                    process.kill(parentPid, 0);
+                } catch (e) {
```

Comment:
Minor: You can omit `e`, you are not using it.

Reply:
interesting that tsc does not report unused variable